internal/worker/pubsub/remoteserver.go

```diff
@@ -144,9 +144,12 @@ func (r *remoteServer) IntrospectionReport() string {
 }
 
 func (r *remoteServer) onForwarderConnection(topic string, details forwarder.OriginTarget, err error) {
+	ctx, cancel := r.scopedContext()
+	defer cancel()
```

Comment:
Should this not be passed in?

Reply:
No, the pubsub handlers/callbacks doesn't take a context, you have to spawn a new one.